### ISSUE TITLE
NBT bugs + No tests

### ISSUE THREAD

**Issue by toqueteos** [PistonDevelopers/hematite_server#18]
**EDIT:** I found http://minecraft.gamepedia.com/NBT_format a lot more useful than http://wiki.vg/NBT.

I was doing some manual testing for NBT comparing with NBTExplorer's output and found that compounds are not written correctly.

Example:

``` rust
let mut c: Compound = HashMap::new();
c.insert("name".to_string(), Nbt::String("Herobrine".to_string()));
let nbt = Nbt::Compound(c);
let mut dst = Vec::new();
<Nbt as Protocol>::proto_encode(&nbt, &mut dst).unwrap();
```

This should return:

```
0a,
    00, 00,                                    // compound length?
    08,                                        // value type
    00, 04,                                    // key name length
    110, 97, 109, 101,                         // key name
    00, 09,                                    // value name length
    72, 101, 114, 111, 98, 114, 105, 110, 101, // value
00
```

But instead it returns:

```
0a,
    00, 00, 00, 01,                            // compound length
    00, 04,                                    // key name length
    110, 97, 109, 101,                         // key name
    08,                                        // value type
    00, 09,                                    // value name length
    72, 101, 114, 111, 98, 114, 105, 110, 101, // value
00
```

**Comment by toqueteos** [comment]
Empty lists are quite interesting too.

NBTExplorer:

```
0a                   // compound id
    00 00            // compound length?
    09               // list id
        00 04        // key name length
        6c 69 73 74  // key name: "list"
        01           // list items id
        00 00 00 00  // prefix length
00                   // compound end
```

nbt.rs:

```
0a              // compound id
    00 00 00 01 // compound length
    00 04       // key name length
    6c 69 73 74 // key name: "list"
    09          // list id
    01          // list items id
    00 00 00 00 // prefix length
00              // compound end
```

**Comment by toqueteos** [comment]
Fixed by #20. _(forgot to add commit message to auto close...)_
